libs/go-libs/sync/shared/sharedmaparr.go

```diff
@@ -0,0 +1,61 @@
+package shared
+
+type SharedMapArr[T any] struct {
+	Shared[map[string]*SharedArr[T]]
```

Comment:
I'm not against generalizing some kind of utility around object sharing, but this start to sound like very specific to a specific use case ^^